fdb-record-layer-core/src/main/proto/record_query_plan.proto

```diff
@@ -905,6 +915,12 @@ message PSimpleComparison {
   optional PComparableObject object = 2;
 }
 
+message PConversionSimpleComparison {
+  optional PComparison.PComparisonType type = 1;
+  optional PComparableObject object = 2;
+  optional KeyExpression conversion = 3;
```

Comment:
It would be somewhat more compact and simpler to just serialize the name of the `QueryableKeyExpression`. But since the `planHash` for the `ConversionXxxComparisons` include the whole `keyExpression`, that would change them, as it would only be possible to get back some nominal prototype instance of the particular function key expression.